src/umap.ts

```diff
@@ -236,6 +283,30 @@ export class UMAP {
     return this.getNEpochs();
   }
 
+  /**
+   * Checks if we're using supervised projection, then process the graph
+   * accordingly.
+   */
+  private processGraphForSupervisedProjection() {
+    const { Y, X } = this;
+    if (Y) {
+      if (Y.length !== X.length) {
+        throw new Error('Length of X and y must be equal');
+      }
+
+      if (this.targetMetric === TargetMetric.categorical) {
+        const lt = this.targetWeight < 1.0;
+        const farDist = lt ? 2.5 * (1.0 / (1.0 - this.targetWeight)) : 1.0e12;
+        this.graph = this.categoricalSimplicialSetIntersection(
+          this.graph,
+          Y,
+          farDist
+        );
+      }
+      // TODO (andycoenen@): add non-categorical supervised embeddings.
```

Comment:
could you add a test that goes through the non-implemented versions and fails/does whatever it should in that case, so that when you add those you have a test to fix?

Reply:
Yup (nothing will happen in this case), but good idea.